etna/models/nn/tft_native/tft.py

```diff
@@ -453,31 +385,45 @@ def step(self, batch: TFTNativeBatch, *args, **kwargs):  # type: ignore
 
         # Get decoder timestamps and pass through gated layer
         decoder_features = self.gated_norm2(
-            x=features[:, -self.decoder_length, :], residual=residual[:, -self.decoder_length, :]
+            x=features[:, -self.decoder_length :, :], residual=residual[:, -self.decoder_length :, :]
         )  # (batch_size, decoder_length, hidden_size)
 
         target_pred = self.output_fc(decoder_features)  # (batch_size, decoder_length, 1)
 
-        loss = self.loss(inputs=target_true, pred=target_pred)
+        return target_pred
+
+    def step(self, batch: TFTNativeBatch, *args, **kwargs):  # type: ignore
+        """Step for loss computation for training or validation.
+
+        Parameters
+        ----------
+        batch:
+            batch of data
+
+        Returns
+        -------
+        :
+            loss, true_target, prediction_target
+        """
+        target_pred = self.forward(batch)  # (batch_size, decoder_length, 1)
+        target_true = batch["decoder_target"].float()  # (batch_size, decoder_length, 1)
+        loss = self.loss(target_pred, target_true)
         return loss, target_true, target_pred
 
     def make_samples(self, df: pd.DataFrame, encoder_length: int, decoder_length: int) -> Iterator[dict]:
         """Make samples from segment DataFrame."""
-        values_target = df["target"].values
-        df["target"] = df["target"].shift(1)
-
         for feature in self.categorical_feature_to_id:
             df[feature] = df[feature].map(self.categorical_feature_to_id[feature])
 
         def _make(
-            values_target: np.ndarray,
             df: pd.DataFrame,
             start_idx: int,
             encoder_length: int,
             decoder_length: int,
         ) -> Optional[dict]:
 
             sample: Dict[str, Any] = {
+                "segment": None,
```

Comment:
Where do you use it? If you use it somewhere you should add it to `TFTNativeBatch` definition